app/controllers/NCTSMessageController.scala

```diff
@@ -16,120 +16,50 @@
-                    case Right(mrn) =>
-                      val processingResult =
-                        saveMessageService.validateXmlSaveMessageUpdateMrn(departure.nextMessageId, xml, messageSender, response, newState, mrn)
-                      processingResult map {
-                        case SubmissionSuccess =>
-                          auditService.auditNCTSMessages(departure.channel, departure.eoriNumber, response, xml)
-                          sendPushNotification(request)
-                          Ok.withHeaders(
-                            LOCATION -> routes.MessagesController.getMessage(departure.departureId, departure.nextMessageId).url
-                          )
-                        case SubmissionFailureInternal =>
-                          val message = "Internal Submission Failure " + processingResult
-                          logger.warn(message)
-                          InternalServerError
-                        case SubmissionFailureExternal =>
-                          val message = "External Submission Failure " + processingResult
-                          logger.warn(message)
-                          BadRequest
-                      }
-                  }
-                case _ =>
-                  val processingResult = saveMessageService.validateXmlAndSaveMessage(request.departure.nextMessageId, xml, messageSender, response, newState)
-                  processingResult map {
-                    case SubmissionSuccess =>
-                      auditService.auditNCTSMessages(departure.channel, departure.eoriNumber, response, xml)
-                      sendPushNotification(request)
-                      Ok.withHeaders(
-                        LOCATION -> routes.MessagesController.getMessage(departure.departureId, departure.nextMessageId).url
-                      )
-                    case SubmissionFailureInternal =>
-                      val message = "Internal Submission Failure " + processingResult
-                      logger.warn(message)
-                      InternalServerError
-                    case SubmissionFailureExternal =>
-                      val message = "External Submission Failure " + processingResult
-                      logger.warn(message)
-                      BadRequest
-                  }
-              }
-            case Left(error) =>
-              Future.successful(BadRequest(error.reason))
+          orchestratorService.saveNCTSMessage(messageSender).map {
+            case Left(_: DepartureAlreadyLocked) => Locked
+            case Left(DepartureNotFound(_))      => Ok
+            case Left(value: ExternalError) =>
+              logger.warn(s"External Submission Failure ${value.reason}")
+              BadRequest(value.reason)
+            case Left(value: models.InternalError) =>
+              logger.warn(s"Internal Submission Failure ${value.reason}")
+              InternalServerError
```

Comment:
Could we possibly enumerate the `ErrorState`s that have exceptions in them here so that we can add the exceptions to the log messages, i.e. `logger.warn(s"Failed to lock departure due to ${value.reason}", value.exception)`? That tends to work better with Kibana / JSON logging as there is a dedicated field for exceptions there.
Then we can remove the exception messages from the `ErrorState#reason` for those errors.